### DISCUSSION TITLE
Request: add a truly unique ID to each channel

### DISCUSSION THREAD

**Discussion by ilikenwf** [iptv-org/community#852] [Ideas]
Right now, there's not a great way of using this list inside a management program without having eventual issues, duplicates, or URL replacement issues, because the only unique data for each entry is URL, tvg-id, and and the title...as such, if a URL simply changes, these tools have trouble identifying which entry changed, and so usually all items with the same name themselves get changed.

**Comment by tarrazuapps** [comment]
My experience after implementing a M3U client is that the true ID for the entries in an M3U is the name.

**Comment by BellezaEmporium** [comment]
Well, if we change that way of working, you may lose the EPG for some programs. It's either this or that.

**Comment by freearhey** [comment]
This was essentially implemented in https://github.com/iptv-org/iptv/pull/20957. So the `tvg-id` should now be unique for each version of each channel.
